modules/ossm-rn-new-features.adoc

```diff
@@ -134,8 +134,20 @@ This release marks the end of support for {SMProductShortName} Control Planes ba
 * Gateway injection is not a supported feature.
 
 === Kubernetes Gateway API
-Kubernetes Gateway API is a technology preview feature that is disabled by default.
+Kubernetes Gateway API is a technology preview feature that is disabled by default. If the Kubernetes API deployment controller is disabled, you must manually deploy and link an ingress gateway to the created Gateway object.
 
+If the Kubernetes API deployment controller is enabled, then an ingress gateway automatically deploys when a Gateway object is created.
+
+==== Installing the Gateway API CRDs
+The Gateway API CRDs do not come pre-installed by default on OpenShift clusters. Install the CRDs prior to enabling Gateway API support in the SMCP.
+
+[source,terminal]
+----
+kubectl get crd gateways.gateway.networking.k8s.io || \
+  { kubectl kustomize "github.com/kubernetes-sigs/gateway-api/config/crd?ref=v0.5.0" | kubectl apply -f -; }
```

Comment:
This command will install Gateway API `v1beta1` that won't work with OSSM 2.2, because it relies on `v1alpha2`.
So maybe we could change it to v0.4.0 and additionally say that OSSM 2.3 will support Gateway API v1beta1 and then github.com/kubernetes-sigs/gateway-api/config/crd?ref=v0.5.0 can be installed or is recommended?

Reply:
@dgn what do you think?

Reply:
@jewertow that CRD contains both versions. It should work with 2.3.0. I can test it in a bit, but I'm pretty sure

Reply:
Yes, it will work with 2.3.0, but won't with 2.2.z.

Reply:
Why? v1beta1 support only comes in 2.3.1

Reply:
Ah, you're right - at least theoretically... It should work with higher version, but I installed Gateway API from tag v0.5.0 and then deployed SMCP v2.2 and I observed a lot of logs in istiod that it tries to list TcpRoute v1alpha2 and errors that it can't list these objects, so I tried to install Gateway API from tag v0.4.0 that brings v1alpha2 and then I didn't get those logs and everything was fine. Unfortunately I didn't download that log. This is basically why I raised my concerns, but you're right - it should work with higher version.

Reply:
@jewertow 2.2 works with the 0.5.0 CRD, see https://github.com/maistra/istio/pull/672

Reply:
Ok, I know why I was getting errors. Look what CRDs are installed using tag v0.5.0:
```
kubectl kustomize "github.com/kubernetes-sigs/gateway-api/config/crd?ref=v0.5.0" | kubectl apply -f - 
customresourcedefinition.apiextensions.k8s.io/gatewayclasses.gateway.networking.k8s.io created
customresourcedefinition.apiextensions.k8s.io/gateways.gateway.networking.k8s.io created
customresourcedefinition.apiextensions.k8s.io/httproutes.gateway.networking.k8s.io created
```
On the other hand, v0.4.0 installs also TcpRoutes and others:
```
kubectl kustomize "github.com/kubernetes-sigs/gateway-api/config/crd?ref=v0.4.0" | kubectl apply -f - 
customresourcedefinition.apiextensions.k8s.io/gatewayclasses.gateway.networking.k8s.io created
customresourcedefinition.apiextensions.k8s.io/gateways.gateway.networking.k8s.io created
customresourcedefinition.apiextensions.k8s.io/httproutes.gateway.networking.k8s.io created
customresourcedefinition.apiextensions.k8s.io/referencepolicies.gateway.networking.k8s.io created
customresourcedefinition.apiextensions.k8s.io/tcproutes.gateway.networking.k8s.io created
customresourcedefinition.apiextensions.k8s.io/tlsroutes.gateway.networking.k8s.io created
customresourcedefinition.apiextensions.k8s.io/udproutes.gateway.networking.k8s.io created
```
This is because Gateway API CRDs has been split into standard and experimental: https://gateway-api.sigs.k8s.io/guides/?h=install#installing-gateway-api.

Reply:
Unfortunately the installation files (experimental-install.yaml and standard-install.yaml) provided by Gateway API community install also some additional deployments like Gateway API admission controller, etc., but it does not start on my OCP cluster, so I would not recommend users to follow that guide.
It's not documented anywhere, but we could install CRDs from tag v0.5.0 without additional controllers with the following command:
```
kubectl kustomize "github.com/kubernetes-sigs/gateway-api/config/crd/experimental?ref=v0.5.0" | kubectl apply -f - 
```

Reply:
@rh-tokeefe we agreed with @dgn that we should recommend to use v0.4.1:
```
kubectl get crd gateways.gateway.networking.k8s.io || \
  { kubectl kustomize "github.com/kubernetes-sigs/gateway-api/config/crd?ref=v0.4.1" | kubectl apply -f -; }
```

Reply:
It will work for both versions: 2.2 and 2.3.